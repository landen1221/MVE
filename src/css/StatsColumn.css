.Stats {
  border: 1px solid black;
  padding: 10px;
}
.Stats h4 {
  text-align: center;
  margin: 3px 0px;
}

.Stats hr {
  margin: 0px 5px;
}

.Stats #intensity-helper {
  font-size: 0.8em;
  margin: 0;
  text-align: center;
  margin-bottom: 10px;
}

.Stats p {
  margin-bottom: 10px;
}
