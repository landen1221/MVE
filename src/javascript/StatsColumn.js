import "../css/StatsColumn.css";

// Example of data provided w/ 'stats'
// {
//   "stats": {
//     "pfizer": 98,
//     "astrazeneca": 84,
//     "moderna": 66.66666666666666,
//     "johnsonandjohnson": 80,
//     "covid": [
//       {
//         "satisfied": "no big deal",
//         "count": "45"
//       },
//       {
//         "satisfied": "mild",
//         "count": "40"
//       },
//         "satisfied": "moderate",
//         "count": "35"
//       },
//       {
//         "satisfied": "severe",
//         "count": "30"
//       },
//     ]
//   }
// }

const StatsColumn = ({ stats, vaccines }) => {
  let totalCovid = 0;
  for (let val of stats["covid"]) {
    totalCovid += parseInt(val["count"]);
  }

  return (
    <>
      <div className="Stats">
        <h4>Satified w/ Vaccine:</h4>
        <hr />
        {Object.entries(vaccines).map(([dbName, siteName]) => (
          <p key={dbName}>
            <b>{siteName === "Johnson & Johnson" ? "J&J" : siteName}: </b>
            {stats[dbName]}%
          </p>
        ))}
      </div>
      <br />
      <div className="Stats">
        <h4>COVID Intensity</h4>
        <hr />
        <p id="intensity-helper">
          (Out of <u>{totalCovid}</u> cases)
        </p>
        {stats["covid"].map((intensity) => (
          <p key={intensity["satisfied"]} className="intensity">
            <b>{intensity["satisfied"]}: </b>{" "}
            {Math.round((intensity["count"] / totalCovid) * 100 * 100) / 100}%
          </p>
        ))}
      </div>
    </>
  );
};

export default StatsColumn;
