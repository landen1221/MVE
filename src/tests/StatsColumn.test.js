import React from "react";
import { render } from "@testing-library/react";
import StatsColumn from "../javascript/StatsColumn";
import { stats, vaccines } from "./globalTestVariables";

// Props: stats, vaccines

it("renders without crashing", () => {
  render(<StatsColumn stats={stats} vaccines={vaccines} />);
});
