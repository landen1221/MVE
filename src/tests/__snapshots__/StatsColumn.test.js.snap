// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Stats"
  >
    <h4>
      Satified w/ Vaccine:
    </h4>
    <hr />
    <p
      id="intensity-helper"
    >
      (From 
      <u>
        5
      </u>
       users)
    </p>
    <p
      class="satisfied"
    >
      <b>
        J&J: 
      </b>
      100%
    </p>
    <p
      class="satisfied"
    >
      <b>
        AstraZeneca: 
      </b>
      100%
    </p>
    <p
      class="satisfied"
    >
      <b>
        Pfizer: 
      </b>
      100%
    </p>
    <p
      class="satisfied"
    >
      <b>
        Moderna: 
      </b>
      50%
    </p>
  </div>
  <br />
  <div
    class="Stats"
  >
    <h4>
      COVID Intensity
    </h4>
    <hr />
    <p
      id="intensity-helper"
    >
      (Out of 
      <u>
        5
      </u>
       cases)
    </p>
    <p
      class="intensity"
    >
      <b>
        No Big Deal: 
      </b>
      20%
    </p>
    <p
      class="intensity"
    >
      <b>
        Mild: 
      </b>
      20%
    </p>
    <p
      class="intensity"
    >
      <b>
        Severe: 
      </b>
      60%
    </p>
  </div>
</DocumentFragment>
`;
